{
  "uid" : "c6eda87d073df44b",
  "name" : "Verify redirection to used resources page not authorized user.",
  "fullName" : "tests.main_page_test.TestMainPage#test_mp_08_verify_redirection_to_used_resources_page",
  "historyId" : "43df35c0ef3ab6719310f8b9b9ee7530",
  "time" : {
    "start" : 1735783067778,
    "stop" : 1735783108255,
    "duration" : 40477
  },
  "status" : "broken",
  "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
  "statusTrace" : "self = <pages.main_page.MainPage object at 0x7f4af9a54260>\nurl = 'https://brainup.site/used-resources'\n\n    def check_expected_link(self, url):\n        with allure.step(f'Check url is present: {url}'):\n            try:\n>               return Wait(self.driver, self.timeout).until(\n                    ec.url_to_be(url), message=f\"Can't find element by locator {url}\")\n\npages/base_page.py:68: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"64027b0abaf9834153ee3da5dac71a29\")>\nmethod = <function url_to_be.<locals>._predicate at 0x7f4af9a63f60>\nmessage = \"Can't find element by locator https://brainup.site/used-resources\"\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources\n\n/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <tests.main_page_test.TestMainPage object at 0x7f4af9c15610>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"64027b0abaf9834153ee3da5dac71a29\")>\nmain_page_open = None\n\n    @allure.title(\"Verify redirection to used resources page not authorized user.\")\n    def test_mp_08_verify_redirection_to_used_resources_page(self, driver, main_page_open):\n        page = MainPage(driver)\n>       resources_page_url = page.open_used_resources_page()\n\ntests/main_page_test.py:57: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:68: in open_used_resources_page\n    self.check_expected_link(MainPageLinks.URL_USED_RESOURCES_PAGE)\npages/base_page.py:72: in check_expected_link\n    return Wait(self.driver, self.timeout).until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"64027b0abaf9834153ee3da5dac71a29\")>\nmethod = <function url_to_be.<locals>._predicate at 0x7f4af9a63740>\nmessage = \"Can't find element by locator https://brainup.site/used-resources\"\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources\n\n/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "main_page_open",
    "time" : {
      "start" : 1735783064144,
      "stop" : 1735783067777,
      "duration" : 3633
    },
    "status" : "passed",
    "steps" : [ {
      "name" : "Open page: https://brainup.site/",
      "time" : {
        "start" : 1735783064144,
        "stop" : 1735783067777,
        "duration" : 3633
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "driver",
        "value" : "<selenium.webdriver.chrome.webdriver.WebDriver (session=\"64027b0abaf9834153ee3da5dac71a29\")>"
      } ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 1,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  }, {
    "name" : "clear_allure_results_folder",
    "time" : {
      "start" : 1735782101571,
      "stop" : 1735782101571,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "driver",
    "time" : {
      "start" : 1735783063386,
      "stop" : 1735783064144,
      "duration" : 758
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "status" : "broken",
    "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
    "statusTrace" : "self = <pages.main_page.MainPage object at 0x7f4af9a54260>\nurl = 'https://brainup.site/used-resources'\n\n    def check_expected_link(self, url):\n        with allure.step(f'Check url is present: {url}'):\n            try:\n>               return Wait(self.driver, self.timeout).until(\n                    ec.url_to_be(url), message=f\"Can't find element by locator {url}\")\n\npages/base_page.py:68: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"64027b0abaf9834153ee3da5dac71a29\")>\nmethod = <function url_to_be.<locals>._predicate at 0x7f4af9a63f60>\nmessage = \"Can't find element by locator https://brainup.site/used-resources\"\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources\n\n/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <tests.main_page_test.TestMainPage object at 0x7f4af9c15610>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"64027b0abaf9834153ee3da5dac71a29\")>\nmain_page_open = None\n\n    @allure.title(\"Verify redirection to used resources page not authorized user.\")\n    def test_mp_08_verify_redirection_to_used_resources_page(self, driver, main_page_open):\n        page = MainPage(driver)\n>       resources_page_url = page.open_used_resources_page()\n\ntests/main_page_test.py:57: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:68: in open_used_resources_page\n    self.check_expected_link(MainPageLinks.URL_USED_RESOURCES_PAGE)\npages/base_page.py:72: in check_expected_link\n    return Wait(self.driver, self.timeout).until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"64027b0abaf9834153ee3da5dac71a29\")>\nmethod = <function url_to_be.<locals>._predicate at 0x7f4af9a63740>\nmessage = \"Can't find element by locator https://brainup.site/used-resources\"\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources\n\n/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
    "steps" : [ {
      "name" : "Open used resources page",
      "time" : {
        "start" : 1735783067778,
        "stop" : 1735783108255,
        "duration" : 40477
      },
      "status" : "broken",
      "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/auto-tests-python/auto-tests-python/pages/main_page.py\", line 68, in open_used_resources_page\n    self.check_expected_link(MainPageLinks.URL_USED_RESOURCES_PAGE)\n  File \"/home/runner/work/auto-tests-python/auto-tests-python/pages/base_page.py\", line 72, in check_expected_link\n    return Wait(self.driver, self.timeout).until(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
      "steps" : [ {
        "name" : "Check url is present: https://brainup.site/used-resources",
        "time" : {
          "start" : 1735783067972,
          "stop" : 1735783108255,
          "duration" : 40283
        },
        "status" : "broken",
        "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources\n\n",
        "statusTrace" : "  File \"/home/runner/work/auto-tests-python/auto-tests-python/pages/base_page.py\", line 72, in check_expected_link\n    return Wait(self.driver, self.timeout).until(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
        "steps" : [ ],
        "attachments" : [ ],
        "parameters" : [ ],
        "stepsCount" : 0,
        "attachmentsCount" : 0,
        "shouldDisplayMessage" : true,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 1,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "e0c5c0adc5e9048a",
      "name" : "Screenshot 2025-01-02 01:58:28.361944",
      "source" : "e0c5c0adc5e9048a.png",
      "type" : "image/png",
      "size" : 58734
    } ],
    "parameters" : [ ],
    "stepsCount" : 2,
    "attachmentsCount" : 1,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "driver::0",
    "time" : {
      "start" : 1735783108363,
      "stop" : 1735783108433,
      "duration" : 70
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Main Page."
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "main_page_test"
  }, {
    "name" : "subSuite",
    "value" : "TestMainPage"
  }, {
    "name" : "host",
    "value" : "fv-az1371-454"
  }, {
    "name" : "thread",
    "value" : "2418-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.main_page_test"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 10,
        "skipped" : 0,
        "passed" : 78,
        "unknown" : 0,
        "total" : 88
      },
      "items" : [ {
        "uid" : "48d7fbff38251f05",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/141//#testresult/48d7fbff38251f05",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
        "time" : {
          "start" : 1735697122874,
          "stop" : 1735697163357,
          "duration" : 40483
        }
      }, {
        "uid" : "aef6fee2b589b392",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/140//#testresult/aef6fee2b589b392",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
        "time" : {
          "start" : 1735610352647,
          "stop" : 1735610393098,
          "duration" : 40451
        }
      }, {
        "uid" : "466a4c5edbcb31a6",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/139//#testresult/466a4c5edbcb31a6",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
        "time" : {
          "start" : 1735524091416,
          "stop" : 1735524131890,
          "duration" : 40474
        }
      }, {
        "uid" : "ac3d56de29e25d0c",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/138//#testresult/ac3d56de29e25d0c",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
        "time" : {
          "start" : 1735437966168,
          "stop" : 1735438006624,
          "duration" : 40456
        }
      }, {
        "uid" : "869668786dbf1231",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/137//#testresult/869668786dbf1231",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
        "time" : {
          "start" : 1735350941549,
          "stop" : 1735350982032,
          "duration" : 40483
        }
      }, {
        "uid" : "7da3b9bd38c86f56",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/136//#testresult/7da3b9bd38c86f56",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/used-resources",
        "time" : {
          "start" : 1735264786093,
          "stop" : 1735264826540,
          "duration" : 40447
        }
      }, {
        "uid" : "b12eea9e482ab0b0",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/81//#testresult/b12eea9e482ab0b0",
        "status" : "passed",
        "time" : {
          "start" : 1730512032351,
          "stop" : 1730512032501,
          "duration" : 150
        }
      }, {
        "uid" : "18a5185b222d833c",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/80//#testresult/18a5185b222d833c",
        "status" : "passed",
        "time" : {
          "start" : 1730426208208,
          "stop" : 1730426208355,
          "duration" : 147
        }
      }, {
        "uid" : "96aba1fd8a508c36",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/79//#testresult/96aba1fd8a508c36",
        "status" : "passed",
        "time" : {
          "start" : 1730339452369,
          "stop" : 1730339452518,
          "duration" : 149
        }
      }, {
        "uid" : "5258f23c57c5c837",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/78//#testresult/5258f23c57c5c837",
        "status" : "passed",
        "time" : {
          "start" : 1730252960892,
          "stop" : 1730252961058,
          "duration" : 166
        }
      }, {
        "uid" : "d87bea3b032e3df9",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/77//#testresult/d87bea3b032e3df9",
        "status" : "passed",
        "time" : {
          "start" : 1730166658408,
          "stop" : 1730166658569,
          "duration" : 161
        }
      }, {
        "uid" : "76c99e1e8411a871",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/76//#testresult/76c99e1e8411a871",
        "status" : "passed",
        "time" : {
          "start" : 1730080394094,
          "stop" : 1730080394269,
          "duration" : 175
        }
      }, {
        "uid" : "3094a64270453ebb",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/75//#testresult/3094a64270453ebb",
        "status" : "passed",
        "time" : {
          "start" : 1729994138172,
          "stop" : 1729994138318,
          "duration" : 146
        }
      }, {
        "uid" : "19a0a3116071283e",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/74//#testresult/19a0a3116071283e",
        "status" : "passed",
        "time" : {
          "start" : 1729907131659,
          "stop" : 1729907131803,
          "duration" : 144
        }
      }, {
        "uid" : "81c678a5548117dc",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/73//#testresult/81c678a5548117dc",
        "status" : "passed",
        "time" : {
          "start" : 1729820945498,
          "stop" : 1729820945652,
          "duration" : 154
        }
      }, {
        "uid" : "db1da4eb0c3e39ec",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/72//#testresult/db1da4eb0c3e39ec",
        "status" : "passed",
        "time" : {
          "start" : 1729734479063,
          "stop" : 1729734479245,
          "duration" : 182
        }
      }, {
        "uid" : "db2c642c2b5bb092",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/71//#testresult/db2c642c2b5bb092",
        "status" : "passed",
        "time" : {
          "start" : 1729648169400,
          "stop" : 1729648169545,
          "duration" : 145
        }
      }, {
        "uid" : "1065216218e8e7df",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/70//#testresult/1065216218e8e7df",
        "status" : "passed",
        "time" : {
          "start" : 1729561860778,
          "stop" : 1729561860932,
          "duration" : 154
        }
      }, {
        "uid" : "9efaa30a9149d773",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/69//#testresult/9efaa30a9149d773",
        "status" : "passed",
        "time" : {
          "start" : 1729475535351,
          "stop" : 1729475535499,
          "duration" : 148
        }
      }, {
        "uid" : "bfe0d972b7c43536",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/68//#testresult/bfe0d972b7c43536",
        "status" : "passed",
        "time" : {
          "start" : 1729389334988,
          "stop" : 1729389335159,
          "duration" : 171
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "c6eda87d073df44b.json",
  "parameterValues" : [ ]
}